targets/TARGET_Silicon_Labs/TARGET_EFM32/crc_api.c

```diff
@@ -109,17 +113,28 @@ void hal_crc_compute_partial(const uint8_t *data, const size_t size)
         return;
     }
 
-    if (((uint32_t)data & 0x3) != 0 || size < 4) {
-        // Unaligned or very small input, run a bytewise CRC
+    if (!enableWordInput || size < 4) {
+        // Input to a non-word-sized poly, or too small data size for a word input
         for (size_t i = 0; i < size; i++) {
             GPCRC_InputU8(GPCRC, data[i]);
         }
     } else {
-        // Aligned input, run 32-bit inputs as long as possible to make go faster.
         size_t i = 0;
-        for (; i < (size & (~0x3)); i+=4) {
+
+        // If input is unaligned, take off as many bytes as needed to align
+        if (((uint32_t)data & 0x3) != 0) {
+            for (; i < (sizeof(uint32_t) - ((uint32_t)data & 0x3)); i++) {
+                GPCRC_InputU8(GPCRC, data[i]);
+            }
+        }
+
+        // If enough input remaining to do word-sized writes, do so
+        while ((size - i) >= sizeof(uint32_t)) {
```

Comment:
Use of `sizeof(uint32_t)` is a little half-hearted and inconsistent - freely mixing with `4` and `0x3` and `i++`.

Don't think you can totally resolve all of that, or that it's worthwhile, but probably shouldn't mix `4` and `sizeof(uint32_t)` in this loop.